pkg/autoupgrade/tags.go

```diff
@@ -35,17 +37,22 @@ func getTagsForImagePattern(ctx context.Context, c daemonClient, namespace, imag
 	return current, append(tags, localTags...), nil
 }
 
-func findLatestTagForImageWithPattern(ctx context.Context, c daemonClient, namespace, image, pattern string) (string, bool, error) {
+func findLatestTagForImageWithPattern(ctx context.Context, c daemonClient, current, namespace, image, pattern string) (string, bool, error) {
 	ref, tags, err := getTagsForImagePattern(ctx, c, namespace, strings.TrimSuffix(image, ":"+pattern))
 	if err != nil {
 		return "", false, err
 	}
 
-	newTag := imagename.DefaultTag
+	var newTag string
+	if current == "" {
+		newTag = invalidTag
+	} else {
+		newTag = current
+	}
```

Comment:
nit: Let's simplify this:
```suggestion
	newTag := current
	if newTag == "" {
		newTag = invalidTag
	}
```